Library/Homebrew/cmake/trap_fetchcontent_provider.cmake

```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 3.24) # Dependency providers introduced in CMake 3.24
+
+option(HOMEBREW_ALLOW_FETCHCONTENT "Allow FetchContent to be used in Homebrew builds" OFF)
+
+if (NOT HOMEBREW_ALLOW_FETCHCONTENT)
+    macro(trap_fetchcontent_provider method depName)
+        message(FATAL_ERROR "Refusing to populate dependency '${depName}' with FetchContent while building in Homebrew, please use a formula dependency or add a resource to the formula.")
+    endmacro()
+
+    cmake_language(
+        SET_DEPENDENCY_PROVIDER trap_fetchcontent_provider
+        SUPPORTED_METHODS FETCHCONTENT_MAKEAVAILABLE_SERIAL
+    )
+endif()
```

Comment:
```suggestion
if (HOMEBREW_ALLOW_FETCHCONTENT)
    return()
endif()

macro(trap_fetchcontent_provider method depName)
    message(FATAL_ERROR "Refusing to populate dependency '${depName}' with FetchContent while building in Homebrew, please use a formula dependency or add a resource to the formula.")
endmacro()

cmake_language(
    SET_DEPENDENCY_PROVIDER trap_fetchcontent_provider
    SUPPORTED_METHODS FETCHCONTENT_MAKEAVAILABLE_SERIAL
)
```
Doesn't work?

Reply:
It does work, see my earlier comment. I don't see early returns often in CMake, but sure we can do it; I'm not particularly opinionated on it one way or another..

Reply:
Ah, sorry; didn't see it since the conversation was marked as resolved. To me reducing indentation always makes things easier to read, but I'm not going to push you on it.

But, re idiomatic CMake, I see lots of early `return`s in the built-in CMake modules (e.g. see scripts in `$(brew --prefix cmake)/share/cmake/Modules`).